### DISCUSSION TITLE
Creating "transition density oribital"

### DISCUSSION THREAD

**Discussion by germanbarcenas** [dftbplus/dftbplus#1161] [Q&A] [answered]
Hello,

I've recently come across the following article ([J. Phys. Chem. A 2011, 115, 44, 12280–12285](https://pubs.acs.org/doi/10.1021/jp203826q)). In it, the authors produce Figure 7 to show the "transition probability density" using their Equation 6.

$\rho(x,y,z)= |\phi_{HOMO}(r)|x|\phi_{LUMO}(r)|^2+\phi_{HOMO}(r)|y|\phi_{LUMO}(r)|^2+\phi_{HOMO}(r)|z|\phi_{LUMO}(r)|^2$
 
It also looks like they have used DFTB+ to produce this image. I would like to produce a similar image for my own work. I'm reading through the waveplot documentation and I'd like to ask this group how to translate Eq. 6 into the waveplot info. 

My understanding is that the "ChargeDensity" keyword must be selected as true, and the selected orbitals should be the HOMO and LUMO orbital. I'm not sure about the order of operations happening with the "ChargeDensity" keyword to know if if would actually produce Eq. 6. I am new to this software, so I would appreciate any help. 

Thank you,

**Comment by germanbarcenas** [comment]
Hello,

Just pinging this again for some attention.

**Comment by aradi** [comment]
We try to reach out to the authors to "donate" their script which created that figures. But that could take a while. You might also try to contact them directly. I think, creating those figures should not be too complicated also yourself. You could created Gaussian cube files with waveplot for HOMO and LUMO and the process them (e.g. in Python) to deliver the necessary information. As the cube file contains the description of the grid as well as the volumetric data, it should be not too complicated to generate a new cube file. (Actually, in `cubemanip` we have some ancient cube-reader, in case it is useful for you.)

**Comment by bhourahine** [comment]
Unfortunately, it is not exactly what you are asking for, but there is also
now material covering plotting the transition densities and orbitals from
the ground state to specified excited states in the DFTB+ recipes:
https://dftbplus-recipes.readthedocs.io/en/latest/linresp/naturalorbitals.html

**Comment by aradi** [comment, marked as answer]
OK, I've modified our old `cubemanip` script, so that it can handle also positions. You find the modified version at https://github.com/aradi/dftbplus/blob/cubemanip/tools/misc/cubemanip, the updated version will become part of the official distribution at some point. You can use the functions `posx()`, `posy()` and `posz()` to get the grid positions. For example, using
```
cubemanip '(($1) * posx() * ($2))**2 + (($1) * posy() * ($2))**2 + (($1) * posz() * ($2))**2' wp-1-1-4-real.cube wp-1-1-5-real.cube > test.cube
```
I obtained for the H2O HOMO-LUMO (from the Waveplot recipes example) the attached figure with an isovalue of 1e-3. This should correspond to the quantity you wish to plot (I hope).

![test](https://user-images.githubusercontent.com/2452321/222190746-00fe7733-d34b-4681-9d3b-2a4ce2efb12c.jpg)

**Reply by charlymedrano7** [reply]
Hi everyone!

Thanks Bálint for this fast solution to the question and the modified version of cubemanip. I think your results for the molecule of water make sense assuming the transition density between HOMO and LUMO as in the equation depicted by the authors. If one plot the orbitals involved one should be able to see the regions of the space in which they are overlapped contributing to the transition density. I reproduce your results for the water molecule and extended it to a Zn tetraporphyrin (ZnTPP, which is  different to the ones used in the cited article but at least has a similar structure). In a comparison with their respective HOMO and LUMO orbitals, the obtained transition densities look good. It seems like the script is working fine.
![image](https://user-images.githubusercontent.com/12562219/224021586-235bd083-dc91-440c-9b88-2dd9a2bd3e50.png)

**Comment by germanbarcenas** [comment]
Thank you so much! It's very generous of you. I had actually already tried
to contact the author but I was not getting a response, so I came here
where I saw some of the coauthors of that paper. I think this should be
enough for me to continue on my own.
